internal/service/networkmanager/vpn_attachment.go

```diff
@@ -0,0 +1,323 @@
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+		},
+	}
+}
+
+func resourceVpnAttachmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	coreNetworkID := d.Get("core_network_id").(string)
+	vpnARN := d.Get("vpn_arn").(string)
+	input := &networkmanager.CreateSiteToSiteVpnAttachmentInput{
+		CoreNetworkId:    aws.String(coreNetworkID),
+		VpnConnectionArn: aws.String(vpnARN),
+	}
+
+	if len(tags) > 0 {
+		input.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	log.Printf("[DEBUG] Creating Network Manager VPN Attachment: %s", input)
+	output, err := conn.CreateSiteToSiteVpnAttachmentWithContext(ctx, input)
+
+	if err != nil {
+		return diag.Errorf("creating Network Manager VPN (%s) Attachment (%s): %s", vpnARN, coreNetworkID, err)
+	}
+
+	d.SetId(aws.StringValue(output.SiteToSiteVpnAttachment.Attachment.AttachmentId))
+
+	if _, err := waitVpnAttachmentCreated(ctx, conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
+		return diag.Errorf("waiting for Network Manager VPN Attachment (%s) create: %s", d.Id(), err)
+	}
+
+	return resourceVpnAttachmentRead(ctx, d, meta)
+}
+
+func resourceVpnAttachmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
+
+	vpnAttachment, err := FindVpnAttachmentByID(ctx, conn, d.Id())
+
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] Network Manager VPN Attachment %s not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return diag.Errorf("reading Network Manager VPN Attachment (%s): %s", d.Id(), err)
+	}
+
+	a := vpnAttachment.Attachment
+	arn := arn.ARN{
+		Partition: meta.(*conns.AWSClient).Partition,
+		Service:   "networkmanager",
+		AccountID: meta.(*conns.AWSClient).AccountID,
+		Resource:  fmt.Sprintf("attachment/%s", d.Id()),
+	}.String()
+	d.Set("arn", arn)
+	d.Set("attachment_policy_rule_number", a.AttachmentPolicyRuleNumber)
+	d.Set("attachment_type", a.AttachmentType)
+	d.Set("core_network_arn", a.CoreNetworkArn)
+	d.Set("core_network_id", a.CoreNetworkId)
+	d.Set("edge_location", a.EdgeLocation)
+	d.Set("owner_account_id", a.OwnerAccountId)
+	d.Set("resource_arn", a.ResourceArn)
+	d.Set("segment_name", a.SegmentName)
+	d.Set("state", a.State)
+	d.Set("vpn_arn", a.ResourceArn)
+
+	tags := KeyValueTags(a.Tags).IgnoreAWS().IgnoreConfig(ignoreTagsConfig)
+
+	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
+		return diag.Errorf("Setting tags: %s", err)
```

Comment:
We prefer lowercase, since error messages get composed together

```suggestion
		return diag.Errorf("setting tags: %s", err)
```